libraries-testing/src/test/java/com/baeldung/dbunit/DataSourceDBUnitTest.java

```diff
@@ -0,0 +1,122 @@
+package com.baeldung.dbunit;
+
+import org.dbunit.Assertion;
+import org.dbunit.DataSourceBasedDBTestCase;
+import org.dbunit.assertion.DiffCollectingFailureHandler;
+import org.dbunit.assertion.Difference;
+import org.dbunit.dataset.IDataSet;
+import org.dbunit.dataset.ITable;
+import org.dbunit.dataset.xml.FlatXmlDataSetBuilder;
+import org.dbunit.operation.DatabaseOperation;
+import org.h2.jdbcx.JdbcDataSource;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.platform.commons.logging.Logger;
+import org.junit.platform.commons.logging.LoggerFactory;
+
+import java.io.InputStream;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import static com.baeldung.dbunit.ConnectionSettings.JDBC_URL;
+import static java.util.stream.Collectors.joining;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class DataSourceDBUnitTest extends DataSourceBasedDBTestCase {
+
+    private static final Logger logger = LoggerFactory.getLogger(DataSourceDBUnitTest.class);
+
+    @Override
+    protected javax.sql.DataSource getDataSource() {
+        JdbcDataSource dataSource = new JdbcDataSource();
+        dataSource.setURL(JDBC_URL);
+        dataSource.setUser("sa");
+        dataSource.setPassword("");
+        return dataSource;
+    }
+
+    @Override
+    protected IDataSet getDataSet() throws Exception {
+        try (InputStream resourceAsStream = getClass().getClassLoader().getResourceAsStream("dbunit/data.xml")) {
+            return new FlatXmlDataSetBuilder().build(resourceAsStream);
+        }
+    }
+
+    @Override
+    protected DatabaseOperation getSetUpOperation() {
+        return DatabaseOperation.REFRESH;
+    }
+
+    @Override
+    protected DatabaseOperation getTearDownOperation() {
+        return DatabaseOperation.DELETE_ALL;
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        super.tearDown();
+    }
+
+    @Test
+    public void testSimpleDataSet() throws SQLException {
```

Comment:
one last thing on the PR, sorry I didn't notice before. We need to follow the test naming conventions: givenX_whenY_thenZ e.g.: whenSendingAPost_thenCorrectStatusCode more info on baeldung - for authors - 2 - code in the article

if there are any tests in the article we need to update it there as well

Reply:
just updated them.